lib/transcoder/audio.ex

```diff
@@ -105,52 +119,36 @@ defmodule Membrane.Transcoder.Audio do
     builder
   end
 
-  defp maybe_plug_resampler(builder, input_format, %Opus{}) do
-    if Map.get(input_format, :sample_rate) == @opus_sample_rate do
-      builder
-    else
-      builder
-      |> child(:resampler, %Membrane.FFmpeg.SWResample.Converter{
-        output_stream_format: %RawAudio{
-          sample_format: :s16le,
-          sample_rate: @opus_sample_rate,
-          channels: Map.get(input_format, :channels, 1)
-        }
-      })
-    end
-  end
-
-  defp maybe_plug_resampler(builder, input_format, %AAC{}) do
-    if Map.get(input_format, :sample_rate) in @aac_sample_rates do
-      builder
-    else
-      builder
-      |> child(:resampler, %Membrane.FFmpeg.SWResample.Converter{
-        output_stream_format: %RawAudio{
-          sample_format: :s16le,
-          sample_rate: 44_100,
-          channels: Map.get(input_format, :channels, 1)
-        }
-      })
-    end
-  end
-
-  defp maybe_plug_resampler(
-         builder,
-         input_format,
-         %MPEGAudio{}
-       ) do
-    if Map.get(input_format, :sample_rate) == @mpeg_raw_audio_format.sample_rate and
-         Map.get(input_format, :sample_format) == @mpeg_raw_audio_format.sample_format and
-         Map.get(input_format, :channels) ==
-           @mpeg_raw_audio_format.channels do
-      builder
-    else
-      builder
-      |> child(:resampler, %Membrane.FFmpeg.SWResample.Converter{
-        output_stream_format: @mpeg_raw_audio_format
-      })
-    end
+  defp maybe_plug_resampler(builder, input_format, %Opus{})
+       when not is_opus_compliant(input_format) do
+    builder
+    |> child(:resampler, %Membrane.FFmpeg.SWResample.Converter{
+      output_stream_format: %RawAudio{
+        sample_format: :s16le,
+        sample_rate: 48_000,
+        channels: 1
+      }
+    })
+  end
+
+  defp maybe_plug_resampler(builder, input_format, %AAC{})
+       when not is_aac_compliant(input_format) do
+    builder
+    |> child(:resampler, %Membrane.FFmpeg.SWResample.Converter{
+      output_stream_format: %RawAudio{
+        sample_format: :s16le,
+        sample_rate: 44_100,
+        channels: 1
+      }
+    })
+  end
+
+  defp maybe_plug_resampler(builder, input_format, %MPEGAudio{})
+       when not is_mp3_compliant(input_format) do
+    builder
+    |> child(:resampler, %Membrane.FFmpeg.SWResample.Converter{
+      output_stream_format: %RawAudio{sample_rate: 44_100, sample_format: :s32le, channels: 2}
+    })
```

Comment:
Looks very nice, but are you sure we don't want to plug resampler if e.g. input format doesn't have `:channels` field?